javascript/javascript2/week2/script.js

```diff
@@ -0,0 +1,145 @@
+{"title": "10,000 BC","year": 2008,"rating": 5.1,"votes": 114750,"running_times": 6540},
+{"title": "100 Feet","year": 2008,"rating": 4.0,"votes": 10979,"running_times": 6240},
+{"title": "100 Girls","year": 2000,"rating": 5.9,"votes": 19232,"running_times": 5640},
+{"title": "101 Dalmatians","year": 1996,"rating": 5.7,"votes": 83245,"running_times": 6180}];
+
+
+
+//1-short movies titles
+const filteredShortMoviesTitle= movies
+.map((oneTitle)=>oneTitle.title)
+.filter((oneTitle)=> oneTitle.length<=13);
+
+console.log(`Movies with a short title are: ${filteredShortMoviesTitle}`); // output Movies with a short title are: '71,'A' gai wak,10,10 Years,10,000 BC,100 Feet,100 Girls
+
+//2-long movies titles
+const filteredLongMoviesTitle =movies
+.map((oneTitle)=>oneTitle.title)
+.filter((oneTitle)=> oneTitle.length>13);
+
+console.log(`Movies with a long title are: ${filteredLongMoviesTitle}`); // output Movies with a long title are: 'Breaker' Morant,'Crocodile' Dundee II,(500) Days of Summer,*batteries not included,...E tu vivrai nel terrore! L'aldilà,...and justice for all.,10 Cloverfield Lane,10 Items or Less,10 Things I Hate About You,101 Dalmatians
+
+//3-movies made between 1980-1989
+
+function filterYearListings (movies, filterObjectYears){
+const minYear =filterObjectYears.minYear;
+const maxYear =filterObjectYears.maxYear;
+return movies.
+filter((oneYear)=>oneYear.year<=maxYear 
+&& oneYear.year>= minYear
+
+
+);}
+
+const filterObjectYears ={
+    minYear: 1980,
+    maxYear:1989, 
+};
+
+const filteredYears =filterYearListings(movies, filterObjectYears);
+console.log(`Between the years 1980-1989 there are ${filteredYears.length} movies made.`); // output Between the years 1980-1989 there are 6 movies made. 
+
+//4-extra key called tag
+
+const newMovieArrayWithNewTag = movies
+.map((newKey) =>{
+    return newKey.rating >= 7 ? {...newKey, tag:"Good"} :
+    newKey.rating >= 4 && newKey.rating < 7 ? {...newKey, tag:"Average"}:
+    newKey.rating < 4 ? {...newKey, tag:"Bad"}:  newKey
+});
+
+console.log(newMovieArrayWithNewTag); //The map() function won't modify the initial array, but creates a new one
+console.log(movies); 
+
+//5-using chaining
+
+const filtredRatedMovies = newMovieArrayWithNewTag
+.filter((oneMovie)=>oneMovie.rating>6 )
+.map((oneMovie) => oneMovie.rating);
+
+console.log(filtredRatedMovies); //output [7.4, 7.9, 7.7, 6.1, 6.7, 7.2]
+
+
+//6- 
+// using match method
+
+const filteredMovieWords =movies
+.filter( (oneMovie)=>
+oneMovie.title.match(/Alien|100|Surfer/i)
+);
+console.log(filteredMovieWords);
+
+//7- duplicated titles
+
+const moviesWithDuplicatedTitle = movies
+.filter((movie) => {
+const strToArr = movie.title.toLocaleLowerCase().split(' ').sort();
+for (let i = 0; i < strToArr.length; i++) {
+if (strToArr[i] === strToArr[i + 1]) {
+return movie;
+}
+}
+ });
+ console.log(moviesWithDuplicatedTitle); //not sure how to approash it, Im getting an empty array 
```

Comment:
You did right 👍 , you just do not have any movies with repeated word inside.
Try to add a repeated word to some movie titles then those movies will be returned.